hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java

```diff
@@ -124,18 +124,21 @@ public List<DataConnectionResource> listResources() {
         try (Connection connection = getConnection()) {
             DatabaseMetaData metaData = connection.getMetaData();
             //Retrieving the columns in the database
-            ResultSet tables = metaData.getTables(null, null, "%", null);
-            List<DataConnectionResource> result = new ArrayList<>();
-            while (tables.next()) {
-                String[] name = Stream.of(tables.getString("TABLE_CAT"),
-                                tables.getString("TABLE_SCHEM"),
-                                tables.getString("TABLE_NAME"))
-                        .filter(Objects::nonNull)
-                        .toArray(String[]::new);
-
-                result.add(new DataConnectionResource(OBJECT_TYPE_TABLE, name));
+            try (ResultSet tables = metaData.getTables(null, null, "%", null)) {
```

Comment:
Maybe merge the two try blocks into one?

Reply:
DatabaseMetaData is not of type AutoCloseable. It won't compile

Reply:
I meant the following:
```
try (Connection connection = getConnection();
    ResultSet tables = connection.getMetaData().getTables(null, null, "%", null)) {
```

Reply:
rev : 4a56eaea6902b349ecb5946bb7f2231d4c45f013